pypa/pypa.go

```diff
@@ -0,0 +1,125 @@
+package pypa
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/aquasecurity/vuln-list-update/types"
+	"github.com/aquasecurity/vuln-list-update/utils"
+	"github.com/cheggaaa/pb"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	pypaDir            = "pypa"
+	securityTrackerURL = "https://github.com/pypa/advisory-db/archive/refs/heads/main.zip"
+	retry              = 3
+	yamlExt            = ".yaml"
+)
+
+type options struct {
+	url   string
+	dir   string
+	retry int
+}
+type option func(*options)
+
+type Pypa struct {
+	opts  *options
+	AppFs afero.Fs
+}
+
+func WithURL(url string) option {
+	return func(opts *options) { opts.url = url }
+}
+
+func WithDir(dir string) option {
+	return func(opts *options) { opts.dir = dir }
+}
+
+func WithRetry(retry int) option {
+	return func(opts *options) { opts.retry = retry }
+}
+
+func NewPypa(opts ...option) Pypa {
+	o := &options{
+		url:   securityTrackerURL,
+		dir:   filepath.Join(utils.VulnListDir(), pypaDir),
+		retry: retry,
+	}
+
+	for _, opt := range opts {
+		opt(o)
+	}
+
+	return Pypa{
+		opts:  o,
+		AppFs: afero.NewOsFs(),
+	}
+
+}
+
+func (pypa *Pypa) Update() error {
+	dir, err := utils.DownloadToTempDir(context.Background(), pypa.opts.url)
+
+	if err != nil {
+		return xerrors.Errorf("failed to download %s: %w", pypa.opts.url, err)
+	}
+
+	vulnDir := filepath.Join(dir, "advisory-db-main/vulns")
+
+	yamlFiles, err := getYamlFiles(vulnDir)
+
+	if err != nil {
+		return xerrors.Errorf("failed to find vulnerability files in the directory %s: %w", vulnDir, err)
+	}
+
+	bar := pb.StartNew(len(yamlFiles))
+
+	for _, file := range yamlFiles {
+		data, err := ioutil.ReadFile(file)
```

Comment:
ioutil is deprecated
```suggestion
		data, err := os.ReadFile(file)
```